### ISSUE TITLE
Table loading methods other than load_table require table to exist

### ISSUE THREAD

**Issue by randyzwitch** [omnisci/pymapd#185]
It's a bit confusing that we export all of the different `load_table_*` methods, yet only `load_table` has the method for auto-creating table.

Moving the auto-create statements into the individual methods would make it easier for users to call the different methods. Alternatively, we could not export anything other than `load_table`, have the individual methods be private(-ish)

**Comment by jp-harvey** [comment]
In my mind it's not that confusing, the `load_table` is a wrapper load function that would be used in the majority of cases, and the others are called by `load_table` downstream. You can still specify a load type with `load_table` and also specify whether or not to create the table. The downstream functions assume that the table is already there so it would be a potentially breaking change, and not sure it really removes much / any potential confusion?

**Comment by randyzwitch** [comment]
Yeah, it feels like to me the better route would be to deprecate the other methods, making people always go through `load_table`. Trade-off between breaking things and cleaning up the API.
